test/diff/helper.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diff
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"testing"
+)
+
+// ErrorIfWantNotGot Wrap cmp.Diff with a t.Helper(). It takes the testing.T
+// want,got, format message and cmp.option as an argument. Therefore, if diff was found
+// it will call t.Error of format and diff that wrapped by PrintWantGot func
+func ErrorIfWantNotGot(t *testing.T, want interface{}, got interface{}, format string, opts ...cmp.Option) {
```

Comment:
Maybe it would be easier to just have one function and allow the user fo pass in the `t.Fatalf`or `t.Error`to the function. I think you could pass it in like this: 
```
FailIfWantGot( t *testing.T, want, got interface{}, format string, errorFunc func(string, ...interface{}), opts ...cmp.Option)
```

Reply:
Thanks for the suggestion @NikeNano  😁
I have some questions, is that okay to take many arguments? 
since testing.T is getting passed we can call it directly, but we have to separate the func.

Reply:
What d you mean by ` take many arguments`?

Reply:
@NikeNano I'm not sure what you mean by "it would be easier" here.  Could you maybe say why your approach is easier?  I'm not sure there's so much difference that either solution here is clearly better but I'm interested to hear your view on this.

Reply:
@NikeNano 
what I think is because this is a helper function it should make the function easier to use. and one of the approaches is to reduce the argument of the function. since testing.T is getting passed we can call it directly.